### DISCUSSION TITLE
Auto generated templates

### DISCUSSION THREAD

**Discussion by davidjgraph** [jgraph/drawio#3414] [General]
Short version: Go to https://app.diagrams.net/?smart-template=1 , enter a diagram description and press "generate".

We're playing with auto generated templates from natural text. That's found in the templates dialog which is shown when you create a new diagram, or in the insert menu / "+" toolbar pulldown:

<img width="214" alt="Capture d’écran 2023-03-08 à 21 01 20" src="https://user-images.githubusercontent.com/1761087/223849062-81810e77-04e4-48f3-97fb-43307189b4b5.png">

Select "Template" to get the template dialog up and select the second option "Smart template"

<img width="668" alt="Capture d’écran 2023-03-08 à 20 55 09" src="https://user-images.githubusercontent.com/1761087/223849173-98ce72f3-26b1-4264-8a48-edde56afa52c.png">

Write a line about your diagram in the description field and select the diagram type you want from the pulldown list and click "generate"

<img width="668" alt="Capture d’écran 2023-03-08 à 20 55 09" src="https://user-images.githubusercontent.com/1761087/223849793-dc0d6600-3a11-49a4-a2d9-b8078da1d231.png">

Note it tends to produce the result in English, adding "en français" at the end outputs in French, for example.

You can generate until you're happy with the result. Clicking "insert" adds the template into the current diagram, which may or may not be blank.

The format is a draw.io editable diagram, not an image.

**Comment by davidjgraph** [comment]
Any general views on usability please comment here. If you find a bug raise an issue as normal.

**Reply by JeromeChiampi** [reply]
Hi David, 
Do you plan to add more diagram types? For instance, that would be interesting to generate GUI wireframes with your great feature.

**Reply by davidjgraph** [reply]
These were the types we could manage in the first phase. Wireframing would be a lot harder. It's on our nice to have list, but we don't see any easy way to implement right now.

**Reply by agershun** [reply]
1) Please add more space in the text field with the diagram description
2) Mermaid breaks on ER-diagram with entity names on non-English languages. Please ask chatGPT to transliterate these identificators or translate them in English. This bug is only for the ER Diagram generation.

**Comment by DavidAlexanderM** [comment]
Oh this is cool

**Comment by edcassin** [comment]
How can I add this template in the desktop version?
We use our own chatgpt api keys?

**Reply by davidjgraph** [reply]
You can't the desktop version is designed to be isolated.

**Reply by edcassin** [reply]
I have a python script that returns a mermaid diagram.
It's based on spaCy and is very basic.
Anyone know how I can implement this as a plugin for the desktop version?
This is so I can refine the code it without having to be online. 
My understanding is I need to convert it to javascript and place it in a plugin library.
